### PR TITLE
pr-3

### PR DESCRIPTION
**Pull request by bohdan2704**
added liquebase migration tool, endpoits with put and delete

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -16,8 +16,17 @@
     <properties>
         <maven.checkstyle.plugin.configLocation>checkstyle.xml</maven.checkstyle.plugin.configLocation>
         <java.version>17</java.version>
+        <mapstruct.version>1.5.5.Final</mapstruct.version>
+        <lombok-mapstruct-binding.version>0.2.0</lombok-mapstruct-binding.version>
+        <checkstyle.version>3.3.0</checkstyle.version>
     </properties>
     <dependencies>
+        <!-- ... інші залежності ... -->
```
Comment (MentorMA) [inline pom.xml:24]:
```suggestion
```

File: src/main/java/com/example/bookshop/repository/BookRepository.java [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 import com.example.bookshop.model.Book;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
 
+@Repository
```
Comment (MentorMA) [inline src/main/java/com/example/bookshop/repository/BookRepository.java:7]: Check what will be if you erase this annotation

File: src/main/java/com/example/bookshop/service/BookServiceImpl.java [hunk 3]
```diff
@@ -10,14 +14,38 @@
...
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book bookGotByID = bookRepository.findById(id).orElseThrow(
```
Comment (MentorMA) [inline src/main/java/com/example/bookshop/service/BookServiceImpl.java:47]:
```suggestion
        Book bookById = bookRepository.findById(id).orElseThrow(
```

Comment (MentorMA) [reply src/main/java/com/example/bookshop/service/BookServiceImpl.java:47]: or just book

**Review by MentorMA — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookshop/repository/BookRepository.java (hunk 2 above)
Comment (bohdan2704) [reply src/main/java/com/example/bookshop/repository/BookRepository.java:7]: Nothing happens becouse spring auto recognises repositories that extend JpaRepo

File: src/main/java/com/example/bookshop/dto/BookDto.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class BookDto {
+    private Long id;
+
+    private String title;
```
Comment (okuzan) [inline src/main/java/com/example/bookshop/dto/BookDto.java:12]: let's make use of the features new Java possesses - such classes should be records

File: src/main/java/com/example/bookshop/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/books")
```
Comment (okuzan) [inline src/main/java/com/example/bookshop/controller/BookController.java:19]: extract /api to common project prefix 
`server.servlet.context-path` property can do this

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-book-table.yml
+  - include:
+      file: db/changelog/changes/02-added-is-deleted-column-to-books.yml
+  - include:
+      file: db/changelog/changes/03-added-deleted-column-to-books.yml
+  - include:
+      file: db/changelog/changes/04-removed-is-deleted-column-from-books.yml
```
Comment (okuzan) [inline src/main/resources/db/changelog/db.changelog-master.yaml:9]: always make sure each file ends with exactly one empty line (won't be displayed on GH), if GH shows red sign, it means there's none
image

```suggestion
      file: db/changelog/changes/04-removed-is-deleted-column-from-books.yml
      
```

File: src/main/resources/db/changelog/changes/03-added-deleted-column-to-books.yml [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+      id: added-deleted-column-to-books
+      author: bohdan
+      changes:
+        - addColumn:
+            tableName: books
+            columns:
+              - column:
+                  name: deleted
+                  type: boolean
+                  defaultValueBoolean: false
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/03-added-deleted-column-to-books.yml:12]:
```suggestion
                  defaultValueBoolean: false
                  
```

File: src/main/resources/db/changelog/changes/02-added-is-deleted-column-to-books.yml [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+      id: added-is-deleted-column-to-books
+      author: bohdan
+      changes:
+        - addColumn:
+            tableName: books
+            columns:
+                - column:
+                    name: is_deleted
+                    type: boolean
+                    defaultValueBoolean: false
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/02-added-is-deleted-column-to-books.yml:12]:
```suggestion
                    defaultValueBoolean: false
                    
```

**Review by okuzan — CHANGES_REQUESTED** [review]

**Comment by bohdan2704** [comment]
i can see commits that i pushed in branch history, but the simply do not appear here in the PR, have this issue for the first time

File: src/main/java/com/example/bookshop/controller/BookController.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+@RequestMapping("/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping
+    public List<BookDto> getAllBooks() {
+        return bookService.findAll();
+    }
+
+    @GetMapping(path = "/{id}")
```
Comment (MentorMA) [inline src/main/java/com/example/bookshop/controller/BookController.java:28]:
```suggestion
    @GetMapping("/{id}")
```

Comment (MentorMA) [reply src/main/java/com/example/bookshop/controller/BookController.java:28]: You can write like this, it`s up to you

**Review by MentorMA — APPROVED** [review]

**Pull request merged**